### ISSUE TITLE
UT99 patch v469c Umod problem

### ISSUE THREAD

**Issue by ProZeidon** [OldUnreal/UnrealTournamentPatches#1144]
Hi - You guys are doing a great job with the patch, but I noticed a bug today:

After install of UT99 patch v469c, every time you want to install a Umod, the product is called "LocalProduct". 
You therefore cannot see what you are installing.

After install of UT99 patch v469c, when you want to uninstall a Umod, then all Umods you have installed after the patch is called LocalProduct. You therefore cannot see what you are uninstalling.

After install of UT99 patch v469c, every time you want to install a Umod, the developer is called the Developer, regardless of what is written inside the Umoden itself.

It has been tested on two different Windows 10 computers with the same result each time.
Umod installs flawlessly before the patch.

Kind regards
ProZeidon

**Comment by SeriousBuggie** [comment]
Can't confirm.
![img](https://user-images.githubusercontent.com/70026933/209012310-80ff994b-ae96-4a22-8808-559ee6fba0e0.png)
I just drag [Setup_Demonica.umod](https://github.com/OldUnreal/UnrealTournamentPatches/files/10281939/Setup_Demonica.zip) into Setup.exe.

Maybe some clues in your Setup.Log?

**Comment by ProZeidon** [comment]
I have tried to drag a Umod into the Setup file, but I get the same result as before.
I can't find anything in the Setup.log that indicates an error with the Umod.

If you want to replicate my error, simply install UT99 from either GoG or Steam.
Then run a registry file so you can install Umods by double-clicking them. Then install an umod or two, to test if it is working (it should be) and then install patch v469c.
From now on, all Umods you install will be called “LocalProduct”.
But the ones you have installed before will be registered correctly.
The only way I have found that can fix this error, is to install a CD version of UT99 in another folder.
It will automatically make Umods work again with the downloaded version of UT99.

In a pathetic and desperate attempt, hoping to find a simple solution to this, that didn't involve a fresh CD installation, I tried copying the Setup.exe file from the CD-ROM version to another computer that also has the same problem, but unfortunately replacing the setup.exe file will make the Setup file not run at all.


The registry file is setup like this:

REGEDIT4

[HKEY_CLASSES_ROOT\.umod]
@="Unreal.Module"

[HKEY_CLASSES_ROOT\Unreal.Module]
@="Unreal Module"

[HKEY_CLASSES_ROOT\Unreal.Module\DefaultIcon]
@="C:\\Program Files (x86)\\GOG Galaxy\\Games\\Unreal Tournament GOTY\\Help\\Unreal.ico"

[HKEY_CLASSES_ROOT\Unreal.Module\Shell]
@="open"

[HKEY_CLASSES_ROOT\Unreal.Module\Shell\open]
@="&Install this Unreal module"

[HKEY_CLASSES_ROOT\Unreal.Module\Shell\open\command]
@="C:\\Program Files (x86)\\GOG Galaxy\\Games\\Unreal Tournament GOTY\\System\\Setup.exe install \"%1\""

**Comment by SeriousBarbie** [comment]
Shouldn't the last registry line be
`@="C:\Program Files (x86)\GOG Galaxy\Games\Unreal Tournament GOTY\System\Setup.exe" install "%1"`

**Comment by ProZeidon** [comment]
It sounded like a reasonable fix.

So I edited it and ran the registry file.
It made no difference.

Therefore I uninstalled the game and ran a cleanup program on the computer.
I downloaded UT GOTY again from GoG and ran the registry file with the new change.
Unfortunately it wouldn't run the Umod files at all now.
So I changed the registration file's text back to the previous one and ran it.
Now the Umod files worked again.
So I ran the patch v469c.
Unfortunately, I have the same problem again. All Umods I want to install are now called "LocalProduct".

**Comment by SeriousBuggie** [comment]
Try use more short patch for store game.
It can be related since v469c use SystemLocalized dir for read .int files now. Which can hit patch limit.

**Comment by ProZeidon** [comment]
After installing the patch I can run a repair of UT through GoG. 
This will reset UT to version v.436. Then I can install Umods correctly again.
When I install the patch again, the Umods I then want to install are registered as "LocalProduct".

**Comment by ProZeidon** [comment]
I experiment with different methods of progress. Hoping it helps track down the problem.
When I install the game from GoG, without doing anything else, I can't install Umods, not even by dragging them into Setup.exe.
If I then install the patch, I suddenly can install Umods. But they are all called LocalProduct.
And I have discovered that the patch also prevents me from uninstalling Umods.
It "pretends" to uninstall, but the files remain and when I open Setup.exe again, the Umods I just uninstalled are there again.

**Comment by SeriousBuggie** [comment]
> When I install the game from GoG, without doing anything else, I can't install Umods, not even by dragging them into Setup.exe.

Because patch allow simple drag with same as call `Setup.exe Your.umod`. Without patch need only old format:
`Setup.exe install Your.umod`. You can do this via command line or make special .bat file near and drag to it.
Bat file content: `Setup.exe install %1`

**Comment by ProZeidon** [comment]
Ah, I didn't know that.
I did another reinstall of the game. This time with a CD version. But Umods didn't work completely correctly with this version either, nor before the patch, which surprised me.. because it has worked with this version all the other times.
So this time I uninstalled again, but instead of using the game's own Setup.exe to uninstall, I used the Windows function to uninstall programs.
The other times I have either uninstalled from GoG Galaxy, Steam or the Setup.exe file.
But when I used Windows' own function, it showed that I had several versions of UT installed, even though the files were gone and the Windows Registry was cleaned.

I removed all of them with the Windows feature and installed the game with GoG, using the basic path that it recommends itself.
Then the same procedure as before.
Now everything works flawlessly.

I apologize that it has been such a big amateur mistake on my part and I appreciate the time you have spent helping.

Yours sincerely.
ProZeidon
